src/main/java/nextstep/subway/line/domain/Line.java

```diff
@@ -53,6 +53,123 @@ public String getColor() {
         return sections;
     }
 
+    public void addSection(Station upStation, Station downStation, int distance) {
+        if (getStations().size() == 0) {
+            sections.add(new Section(this, upStation, downStation, distance));
+            return;
+        }
+        checkSectionAddValidty(upStation, downStation);
+        if (newUpDownStationIsLastStation(upStation, downStation)) {
+            sections.add(new Section(this, upStation, downStation, distance));
+            return;
+        }
+        Section oldSection = findSection(upStation, downStation);
+        int oldDistance = oldSection.getDistance();
+        checkSectionDistanceValidty(oldDistance, distance);
+        removeSection(oldSection.getDownStation().getId());
+        if (oldSection.getUpStation() == upStation) {
+            sections.add(new Section(this, upStation, downStation, distance));
+            sections.add(new Section(this, downStation, oldSection.getDownStation(), oldDistance - distance));
+        }
+        sections.add(new Section(this, oldSection.getUpStation(), upStation, oldDistance - distance));
+        sections.add(new Section(this, upStation, downStation, distance));
+    }
+
+    private void checkSectionDistanceValidty(int oldDistance, int distance) {
+        if (oldDistance <= distance) {
+            throw new RuntimeException("역 사이에 새로운 역을 등록할 경우 기존 역 사이 길이보다 크거나 같으면 등록을 할 수 없음");
+        }
+    }
+
+    private Section findSection(Station upStation, Station downStation) {
+        Section section1 = getSections().stream().filter(it -> it.getUpStation().getId() == upStation.getId()).findFirst().orElse(null);
+        Section section2 = getSections().stream().filter(it -> it.getDownStation().getId() == downStation.getId()).findFirst().orElse(null);
+        if (section1 == null) {
+            return section2;
+        }
+        return section1;
+    }
+
+    private boolean newUpDownStationIsLastStation(Station upStation, Station downStation) {
+        if (getStations().get(0).getId() == downStation.getId()) {
+            return true;
+        }
+        if (getStations().get(getStations().size()-1).getId() == upStation.getId()) {
```

Comment:
띄어쓰기 컨벤션이 지켜지지 않았네요!
Intellij의 Reformat Code 기능을 활용해보세요 :) 

```suggestion
        if (getStations().get(getStations().size() - 1).getId() == upStation.getId()) {
```

Reply:
적용하였습니다 :) 감사합니다!